### PR TITLE
Veto3

### PR DESCRIPTION
**Pull request by siilieva**
- adding Veto3 - detector, support for digitization, raw-data conversion, monitoring, 2dED
- having the Veto pit in the geometry of the TI18
- all backward compatible, including for the testbeam setup

- [x] tested running simulation -> digitization -> 2dED -> monitoring
- [x] tested running raw-data conversion & the  2dED
- [ ]  need more events for a data test on the monitoring  and that would go on a different PR for related changes

The 2024 det. configuration is already on eos: ` /eos/experiment/sndlhc/convertedData/physics/2024/geofile_sndlhc_TI18_V0_2024.root` so that people can run the 2dED when we start the automatic data conversion next week.

### CODE REVIEW

**Review by olantwin — APPROVED** [review]
Generally looks good to me. Some tiny comments.

Might be worth merging a few commits together and splitting some others.

File: shipLHC/scripts/2dEventDisplay.py [hunk 1]
```diff
@@ -31,7 +31,6 @@ def pyExit():
 parser = ArgumentParser()
 parser.add_argument("-r", "--runNumber", dest="runNumber", help="run number", type=int,required=False)
 parser.add_argument("-p", "--path", dest="path", help="path to data file",required=False,default=os.environ["EOSSHIP"]+"/eos/experiment/sndlhc/convertedData/physics/2022/")
-parser.add_argument("-praw", "--pathRaw", dest="pathRaw", help="path to raw data file",required=False,default="/eos/experiment/sndlhc/raw_data/physics/2022/")
```
[the remark this answers is not in the document]
Comment (siilieva) [reply shipLHC/scripts/2dEventDisplay.py:35]: made a separate commit for that change

File: sndFairTasks/boardMappingParser.cxx [hunk 2]
```diff
@@ -118,13 +118,16 @@ tuple<map<string, map<string, map<string, int>> >, map<string, map<string, map<s
...
-        }          
+        }
         // Loop over the slots (the first is always left, the second always right)
         for (int i = 0; i < info.slots.size(); i++)
         {
           bString = Form("board_%i", info.board);
-          if (i==0) boardMapsMu["MuFilter"][bString][info.slots.at(i)] = Form("Veto_%dLeft", stoi(subel.key()));
-          else if (i==1) boardMapsMu["MuFilter"][bString][info.slots.at(i)] = Form("Veto_%dRight", stoi(subel.key()));
+          if (stoi(subel.key())<3){
+            if (i==0) boardMapsMu["MuFilter"][bString][info.slots.at(i)] = Form("Veto_%iLeft", stoi(subel.key()));
```
[the remark this answers is not in the document]
Comment (siilieva) [reply sndFairTasks/boardMappingParser.cxx:127]: I don't remember if we use this conversion to int as a safety net.  I wont touch it. This is ran once per partition and is fast anyways.

**Comment by siilieva** [comment]
I implemented all-1 suggestions, with the the exception for the one about the string-> int conversion in the boarding_parser).

File: sndFairTasks/boardMappingParser.cxx (hunk 2 above)
[the remark this answers is not in the document]
Comment (olantwin) [reply sndFairTasks/boardMappingParser.cxx:127]: I agree that it's something that should not be modified on the fly.

We might want to consider doing some proper error-checking though if we want it as a safety net (see e.g. https://stackoverflow.com/questions/11598990/is-stdstoi-actually-safe-to-use)

**Review by olantwin — APPROVED** [review]
Looks good to me.

**Comment by siilieva** [comment]
thanks a lot  for the review of this large PR, @olantwin !

**Pull request merged**